spec/rule_spec.rb

```diff
@@ -30,6 +30,80 @@
     end
   end
 
+  describe "iterating with a floor_date" do
+    describe "No COUNT or INTERVAL > 1" do
+      it "uses the floor_date provided when iterating" do
+        rrule = 'FREQ=DAILY'
+        dtstart = Time.parse('Tue Sep  2 06:00:00 PDT 1997')
+        timezone = 'America/New_York'
+
+        rrule = RRule::Rule.new(rrule, dtstart: dtstart, tzid: timezone)
+
+        floor_date = Time.parse('Mon Sep  3 06:00:00 PDT 2018')
+
+        expect(rrule.each(floor_date: floor_date).take(3)).to match_array([
+          Time.parse('Tue Sep  3 06:00:00 PDT 2018'),
+          Time.parse('Wed Sep  4 06:00:00 PDT 2018'),
+          Time.parse('Thu Sep  5 06:00:00 PDT 2018')
+        ])
+      end
+    end
+
+    describe "COUNT present" do
+      it "starts at dtstart when iterating" do
+        rrule = 'FREQ=DAILY;COUNT=10'
+        dtstart = Time.parse('Tue Sep  2 06:00:00 PDT 1997')
+        timezone = 'America/New_York'
+
+        rrule = RRule::Rule.new(rrule, dtstart: dtstart, tzid: timezone)
+
+        floor_date = Time.parse('Mon Sep  3 06:00:00 PDT 2018')
+
+        expect(rrule.each(floor_date: floor_date).take(3)).to match_array([
+          Time.parse('Tue Sep  2 06:00:00 PDT 1997'),
```

Comment:
Thanks for this PR - it'll be a huge performance gain for a very common case. My one concern is that `between` continue to work correctly if you pass `floor_date` to a series with a `COUNT` or `INTERVAL` (because it likely wouldn't be obvious to a user why those options are incompatible, and I wouldn't want them to get incorrect results). From looking at the code, it looks like this is the case (i.e. if you pass `floor_date` to `between` for an incompatible RRULE, it'll still work, you just won't get the performance gains) but would you mind adding a spec to confirm that?

Reply:
Added!  https://github.com/square/ruby-rrule/pull/11/commits/14e46ce33e09c709b44fe0d240b831dccfa513f4

Reply:
Thanks for the updated specs - I think I was misunderstanding the code, looks like it behaves as I was hoping. This LGTM. @nerdrew what do you think?